packages/amplify-provider-awscloudformation/lib/system-config-manager.js

```diff
@@ -92,10 +94,11 @@ async function getRoleCredentials(profileConfig) {
     RoleSessionName: 'amplify',
     ExternalId: profileConfig.external_id,
   }).promise();
-  // accessKeyId: data.Credentials.AccessKeyId,
-  // secretAccessKey: data.Credentials.SecretAccessKey,
-  // sessionToken: data.Credentials.SessionToken
-  return roleData.Credentials;
+  return {
+    accessKeyId: roleData.Credentials.AccessKeyId,
+    secretAccessKey: roleData.Credentials.SecretAccessKey,
+    sessionToken: roleData.Credentials.SessionToken,
```

Comment:
Is it okay if the sessionToken is undefined? Did you test that case?

Reply:
no, it's not OK, I tested and it fails without the session token. 
but if the assumeRole is successful, there's always going to be data.Credentials.SessionToken.

Reply:
Okay, so in that case, if my profile doesn't assume an STS role, will the SDK calls fail?

Reply:
If the profile does not assume role, the credentials will be plain old credential with key id and key, there's no session token. 
AWS checks if the credential is a temp credential or a normal credential, for temp credentials, it requires session token.